### ISSUE TITLE
Newman does not execute all tests

### ISSUE THREAD

**Issue by jproceeam** [postmanlabs/newman#949]
Only the first three tests are executed. I would expect them all to be run.

 1. Newman Version (can be found via `newman -v`): 
3.5.0
 2. OS details (type, version, and architecture): 
Windows Server 2012 R2
 3. Are you using Newman as a library, or via the CLI? 
CLI
 3. Did you encounter this recently, or has this bug always been there: 
I have recently started using this software so I can't determine if this bug has always been there
 4. Expected behaviour: 
I expect all tests to be run.
 5. Command / script used to run Newman:
`newman run .\issue_collection`
 6. Sample collection, and auxilliary files (minus the sensitive details):
```json
{
	"variables": [],
	"info": {
		"name": "Issue collection",
		"_postman_id": "e1c5c7d2-1373-3863-2a4a-cd154ec1d3b2",
		"description": "",
		"schema": "https://schema.getpostman.com/json/collection/v2.0.0/collection.json"
	},
	"item": [
		{
			"name": "https://jsonplaceholder.typicode.com/posts/1",
			"event": [
				{
					"listen": "test",
					"script": {
						"type": "text/javascript",
						"exec": [
							"tests[\"Status code is 200\"] = responseCode.code === 200;",
							"tests[\"Content-Type is 'application/json; charset=utf-8'\"] = postman.getResponseHeader(\"Content-Type\") === \"application/json; charset=utf-8\";",
							"",
							"var jsonData = JSON.parse(responseBody);",
							"",
							"// required annotated properties",
							"tests[\"Contains userId\"] = jsonData.userId;",
							"tests[\"Contains something that does not exist\"] = jsonData.doesNotExist;",
							"tests[\"Contains something else that does not exist\"] = jsonData.notExist;",
							"tests[\"Contains more not existing stuff\"] = jsonData.foo;",
							""
						]
					}
				}
			],
			"request": {
				"url": "https://jsonplaceholder.typicode.com/posts/1",
				"method": "GET",
				"header": [],
				"body": {},
				"description": ""
			},
			"response": []
		}
	]
}
```

**Comment by jproceeam** [comment]
Hmm, this does not seem to reproduce the issue. Let me try to cook up something that will trigger this bug.

**Comment by kunagpal** [comment]
@jproceeam Thanks for reporting, we've managed to reproduce this issue. As a temporary workaround, could you use: 
```javascript
tests["Doesn't contain non-existent properties"] = _.isUndefined(jsonData.doesNotExist);
```
to test absence of any given property? (`_` is the in-built Lodash object, documented at https://lodash.com/docs/3.10.1). 

Also, consider checking out the last example on https://www.getpostman.com/docs/testing_examples for an alternative way to validate responses. (The `schema` object in the example can be generated using http://jsonschema.net.

**Comment by kunagpal** [comment]
@jproceeam This issue has been fixed in Newman `v3.5.1`, do go ahead and check it out! :smile:

**Comment by jproceeam** [comment]
Nice! Thank you very much!
